tests/test_database_types.py

```diff
@@ -2,33 +2,117 @@
 import unittest
 import time
 import logging
+import re
+import math
+import datetime
 from decimal import Decimal
+from prestodb.exceptions import PrestoUserError
 
 from parameterized import parameterized
 
 from data_diff import databases as db
 from data_diff.diff_tables import TableDiffer, TableSegment
-from .common import CONN_STRINGS
+from .common import CONN_STRINGS, N_SAMPLES
 
 
-logging.getLogger("diff_tables").setLevel(logging.ERROR)
-logging.getLogger("database").setLevel(logging.WARN)
-
-CONNS = {k: db.connect_to_uri(v) for k, v in CONN_STRINGS.items()}
+CONNS = {k: db.connect_to_uri(v, 1) for k, v in CONN_STRINGS.items()}
 
 CONNS[db.MySQL].query("SET @@session.time_zone='+00:00'", None)
 
-TYPE_SAMPLES = {
-    "int": [127, -3, -9, 37, 15, 127],
-    "datetime_no_timezone": [
-        "2020-01-01 15:10:10",
-        "2020-02-01 9:9:9",
-        "2022-03-01 15:10:01.139",
-        "2022-04-01 15:10:02.020409",
-        "2022-05-01 15:10:03.003030",
-        "2022-06-01 15:10:05.009900",
-    ],
-    "float": [
+class PaginatedTable:
+    # We can't query all the rows at once for large tables. It'll occupy too
+    # much memory.
+    RECORDS_PER_BATCH = 1000000
+
+    def __init__(self, table, conn):
+        self.table = table
+        self.conn = conn
+
+    def __iter__(self):
+        self.last_id = 0
+        self.values = []
+        self.value_index = 0
+        return self
+
+    def __next__(self) -> str:
+        if self.value_index == len(self.values):  #  end of current batch
+            query = f"SELECT id, col FROM {self.table} WHERE id > {self.last_id} ORDER BY id ASC LIMIT {self.RECORDS_PER_BATCH}"
```

Comment:
Oracle doesn't support LIMIT. Instead you have to do `FETCH NEXT n ROWS ONLY`

Reply:
See the line below!

Reply:
Good enough. I'll refactor that later into `sql.py`.